src/tickit/adapters/io/http_io.py

```diff
@@ -1,41 +1,46 @@
 import asyncio
 import logging
-from dataclasses import dataclass
-from inspect import getmembers
-from typing import Awaitable, Callable, Iterable, Optional
+from typing import Awaitable, Callable, Iterable, Optional, Tuple
 
 from aiohttp import web
 from aiohttp.web_routedef import RouteDef
 
-from tickit.adapters.interpreters.endpoints.http_endpoint import HttpEndpoint
-from tickit.core.adapter import Adapter, RaiseInterrupt
-from tickit.core.device import Device
+from tickit.adapters.http import HttpAdapter
+from tickit.adapters.specifications import HttpEndpoint
+from tickit.core.adapter import AdapterIo, RaiseInterrupt
 
 LOGGER = logging.getLogger(__name__)
 
 
-@dataclass
-class HttpAdapter(Adapter):
-    """An adapter implementation which delegates to a server and sets up endpoints.
+class HttpIo(AdapterIo[HttpAdapter]):
+    """An AdapterIo implementation which delegates to a server and sets up endpoints.
 
-    An adapter implementation which delegates the hosting of an http requests to a
-    server and sets up the endpoints for said server.
+    An AdapterIo implementation which delegates the hosting of an http requests to a
+    server and sets up the endpoints for said server from a HttpAdapter.
     """
 
-    host: str = "localhost"
-    port: int = 8080
+    host: str
+    port: int
 
     _stopped: Optional[asyncio.Event] = None
     _ready: Optional[asyncio.Event] = None
```

Comment:
Now that this isn't a dataclass, these could be left unset here, then set to Events in the `__init__` method and then the two `_ensure_x_event` methods aren't needed and everything can use the fields directly

Reply:
I've changed the `None` to be set in the `__init__` instead, but I am weary of potentially ruining the synchronisation by accident if i mess with the rest of it since it was written recently, there is probably some intention behind it.